etl/transform.py

```diff
@@ -62,11 +62,11 @@ def __init__(self, dataFrame: DataFrame):
 
     def transform(self) -> DataFrame:
         with_recommendations = (
-            self.dataFrame.where(~F.col("top_candidates").isNull())
+            self.dataFrame.where(F.size(F.col("top_candidates")) > 0)
```

Comment:
Here you removed a null check, but `top_candidates` schema says the column could be nullable. 
I think your approach here is the correct one, but could we make the schema consistent?

Reply:
Sure!